### PR TITLE
[AV-1057] Midas Voltage Thread

### PR DESCRIPTION
**Pull request by mpkarpov-ui**
This one's pretty simple.. just read voltage from the pin and map it correctly (Used same mapping as TARS)

### CODE REVIEW

File: MIDAS/src/hardware/Voltage.cpp [hunk 1]
```diff
@@ -1,17 +1,19 @@
 #include "sensors.h"
+#include <hal.h>
```
Comment (redindelible) [inline MIDAS/src/hardware/Voltage.cpp:2]: If this include is here just to get `analogRead`, it's probably better to just include `Arduino.h` instead.

**Review by redindelible — CHANGES_REQUESTED** [review]
The voltage sensor thread hasn't been fully implemented in this PR, which is needed to make voltage sensing work.

File: .vscode/c_cpp_properties.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            ],
+            "defines": [
+                "_DEBUG",
+                "UNICODE",
+                "_UNICODE"
+            ],
+            "windowsSdkVersion": "10.0.22621.0",
+            "compilerPath": "cl.exe",
+            "cStandard": "c17",
+            "cppStandard": "c++17",
```
Comment (aidancostello) [inline .vscode/c_cpp_properties.json:16]: This file should be gitignored, can you delete lmao

**Review by aidancostello — CHANGES_REQUESTED** [review]

File: MIDAS/src/hardware/Voltage.cpp [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
 
+/**
+ * Reads the value of the given analog pin and converts it to a battery voltage with the assumption that the voltage sensor is plugged into that pin
+ * \return The scaled voltage given by the voltage sensor
+*/
 Voltage VoltageSensor::read() {
-    // read from aforementioned global instance of sensor
-    return Voltage();
+    Voltage v_battery;
+    v_battery.voltage = analogRead(69) / 1024.f * 3.3f * 3.f;
```
Comment (redindelible) [inline MIDAS/src/hardware/Voltage.cpp:17]: Can you make the 69 into a #define so it's easier to find when we get the real pins

**Review by redindelible — CHANGES_REQUESTED** [review]

File: MIDAS/src/systems.cpp [hunk 4]
```diff
@@ -76,8 +76,8 @@ DECLARE_THREAD(gas, RocketSystems* arg) {
 
 DECLARE_THREAD(voltage, RocketSystems* arg) {
     while (true) {
-        THREAD_SLEEP(16);
-        //Serial.println("VOLT");
+        arg->rocket_state.voltage.update(arg->sensors.voltage.read())
```
Comment (aidancostello) [inline MIDAS/src/systems.cpp:79]: semicolon needed at the end

Comment (mpkarpov-ui) [reply MIDAS/src/systems.cpp]: what python (and not being able to compile) does to a mf

Comment (redindelible) [reply MIDAS/src/systems.cpp]: wait why can't you compile

**Comment by redindelible** [comment]
Could you add .vscode to the .gitignore in the root directory?

**Comment by mpkarpov-ui** [comment]
I believe I've already done that -- I had to git rm --cached it but it shouldnt be being tracked anymore

File: MIDAS/src/systems.cpp (hunk 4 above)
Comment (mpkarpov-ui) [reply MIDAS/src/systems.cpp]: sent the log on slack

**Review by aidancostello — APPROVED** [review]
Don't merge yet but looks good

**Review by redindelible — APPROVED** [review]

**Pull request merged**